### PR TITLE
Hacker Rank Solution

### PR DESCRIPTION
**Pull request by aryabharat**
Some Basic HR solution

**Fixes issue:**

**Changes:**

### CODE REVIEW

File: code/online_challenges/src/Hacker_Rank/3D_Surface_Area/3D_Surface_Area.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
+
+#include <iostream>
+
+int main()
+{
+int h, w;
+    std::cin >> h >> w;
+    int a[h][w];
```
Comment (arnavb) [inline code/online_challenges/src/Hacker_Rank/3D_Surface_Area/3D_Surface_Area.cpp]: C++ does not support VLAs without compiler extensions; use a `std::vector` instead.

File: code/online_challenges/src/Hacker_Rank/3D_Surface_Area/3D_Surface_Area.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
+
+#include <iostream>
+
+int main()
+{
+int h, w;
```
Comment (arnavb) [inline code/online_challenges/src/Hacker_Rank/3D_Surface_Area/3D_Surface_Area.cpp]: Fix indent

File: code/online_challenges/src/Hacker_Rank/3D_Surface_Area/3D_Surface_Area.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    std::cin >> h >> w;
+    int a[h][w];
+    for(int i = 0; i < h; i++)
+    {
```
Comment (arnavb) [inline code/online_challenges/src/Hacker_Rank/3D_Surface_Area/3D_Surface_Area.cpp]: Remove braces

File: code/online_challenges/src/Hacker_Rank/3D_Surface_Area/3D_Surface_Area.cpp [hunk 4]
```diff
@@ -0,0 +1,66 @@
+
+#include <iostream>
+
+int main()
+{
+int h, w;
+    std::cin >> h >> w;
+    int a[h][w];
+    for(int i = 0; i < h; i++)
```
Comment (arnavb) [inline code/online_challenges/src/Hacker_Rank/3D_Surface_Area/3D_Surface_Area.cpp]: Pre-increment.

File: code/online_challenges/src/Hacker_Rank/3D_Surface_Area/3D_Surface_Area.cpp [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        if(i == 0)
+        {
+            for(int j = 0; j < w; j++)
+              {
+                int temp = 0;
+                if(j == 0)
+                  {
+                      temp = (a[i][j] - 1) * 4 + 6;
+                      sum = sum + temp;
```
Comment (arnavb) [inline code/online_challenges/src/Hacker_Rank/3D_Surface_Area/3D_Surface_Area.cpp]:
```suggestion
                      sum += temp;
```

File: code/online_challenges/src/Hacker_Rank/3D_Surface_Area/3D_Surface_Area.cpp [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+                  }
+
+             else
+                 {
```
Comment (arnavb) [inline code/online_challenges/src/Hacker_Rank/3D_Surface_Area/3D_Surface_Area.cpp]: Brace indentation is very badly placed in this file; could you fix this?

File: code/online_challenges/src/Hacker_Rank/3D_Surface_Area/3D_Surface_Area.cpp [hunk 7]
```diff
@@ -0,0 +1,66 @@
+
```
Comment (arnavb) [inline code/online_challenges/src/Hacker_Rank/3D_Surface_Area/3D_Surface_Area.cpp]: Make the `Hacker_Rank` folder called `hackerrank`. Remove this empty line.

File: code/online_challenges/src/Hacker_Rank/3D_Surface_Area/readme [hunk 8]
```diff
@@ -0,0 +1 @@
+Problem Link: https://www.hackerrank.com/challenges/3d-surface-area/problem?h_r=internal-search
```
Comment (arnavb) [inline code/online_challenges/src/Hacker_Rank/3D_Surface_Area/readme]: Please include more than just a link. A markdown file with problem/solution details (like for other online challenges) would be best.

File: code/online_challenges/src/Hacker_Rank/Algo_DP_Maxsubarray/Algo_DP_Maxsubarray.cpp [hunk 9]
```diff
@@ -0,0 +1,55 @@
+#include <iostream>
```
Comment (arnavb) [inline code/online_challenges/src/Hacker_Rank/Algo_DP_Maxsubarray/Algo_DP_Maxsubarray.cpp]: Make sure your file/folder names are all lowercase separated by underscores.

File: code/online_challenges/src/Hacker_Rank/Algo_DP_Maxsubarray/Algo_DP_Maxsubarray.cpp [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+                {
+                    std::cin >> arr[i];
+                    if(arr[i] >= 0)
+
+                    flag++;
+                }
+
+            if (flag == 0)
+                {
+                    sort (arr.begin(), arr.end());
```
Comment (arnavb) [inline code/online_challenges/src/Hacker_Rank/Algo_DP_Maxsubarray/Algo_DP_Maxsubarray.cpp]: Fully qualify this name with `std::`.

File: code/online_challenges/src/Hacker_Rank/Almost_Sorted/Almost_Sorted.cpp [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+    std::vector <int> a(n);
+    std::vector <int> b(n);
+    for(int i; i < n; i++)
+        {
+          std::cin >> a[i];
+          b[i] = a[i];
+        }
+    sort(b.begin(), b.end());
+    int p = 0;
+    int s,l;
```
Comment (arnavb) [inline code/online_challenges/src/Hacker_Rank/Almost_Sorted/Almost_Sorted.cpp]: Just initialize the variables to 0 here.

File: code/online_challenges/src/Hacker_Rank/Almost_Sorted/Almost_Sorted.cpp [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    s = l = 0;
+    int t = 0;
```
Comment (arnavb) [inline code/online_challenges/src/Hacker_Rank/Almost_Sorted/Almost_Sorted.cpp]: Also, better variable names would be nice.

File: code/online_challenges/src/Hacker_Rank/Almost_Sorted/Almost_Sorted.cpp [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+
+        std::cout << "yes" << "\n";
+        std::cout << "reverse " << s + 1 << " " << l + 1;
+    }
+
+
+
+
+
+    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
```
Comment (arnavb) [inline code/online_challenges/src/Hacker_Rank/Almost_Sorted/Almost_Sorted.cpp]: Remove this.

File: code/online_challenges/src/Hacker_Rank/Bigger_is_Greater/Bigger_is_Greater.cpp [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+{
+    int t;
+    std::cin >> t;
+
+    while(t--)
+    {
+        std::string s;
+        std::cin >> s;
+        bool ans = next_permutation(s.begin(), s.end());                       // do permutation and store it in string ans
+        if(ans == false)                                                        // if there is no permutation possible original and ans will be same
```
Comment (arnavb) [inline code/online_challenges/src/Hacker_Rank/Bigger_is_Greater/Bigger_is_Greater.cpp]:
```suggestion
        if(!ans)  // if there is no permutation possible original and ans will be same
```

File: code/online_challenges/src/Hacker_Rank/Bigger_is_Greater/Bigger_is_Greater.cpp [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+int main()
+{
+    int t;
+    std::cin >> t;
+
+    while(t--)
+    {
+        std::string s;
+        std::cin >> s;
+        bool ans = next_permutation(s.begin(), s.end());                       // do permutation and store it in string ans
```
Comment (arnavb) [inline code/online_challenges/src/Hacker_Rank/Bigger_is_Greater/Bigger_is_Greater.cpp]: Comments should be too spaces away from their line.

**Comment by aryabharat** [comment]
Closing for new pull request.

**Pull request closed without merging**